internal/venice-common/src/main/java/com/linkedin/venice/authentication/jwt/TokenAuthenticationService.java

```diff
@@ -0,0 +1,62 @@
+package com.linkedin.venice.authentication.jwt;
+
+import com.linkedin.venice.authentication.AuthenticationService;
+import com.linkedin.venice.authorization.Principal;
+import com.linkedin.venice.utils.VeniceProperties;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * Validates JWT tokens passed as Bearer tokens in the HTTP Authorise Request Header.
+ */
+public class TokenAuthenticationService implements AuthenticationService {
+  private static final Logger log = LogManager.getLogger(TokenAuthenticationService.class);
+  private static final String HTTP_HEADER_VALUE_PREFIX = "Bearer ";
+
+  private AuthenticationProviderToken authenticationProvider;
+
+  @Override
+  public void initialise(VeniceProperties veniceProperties) throws Exception {
+    TokenProperties tokenProperties = new TokenProperties(
+        veniceProperties.getString("authentication.jwt.secretKey", ""),
+        veniceProperties.getString("authentication.jwt.publicKey", ""),
+        veniceProperties.getString("authentication.jwt.authClaim", ""),
+        veniceProperties.getString("authentication.jwt.publicAlg", ""),
+        veniceProperties.getString("authentication.jwt.audienceClaim", ""),
+        veniceProperties.getString("authentication.jwt.audience", ""),
+        veniceProperties.getString("authentication.jwt.jwksHostsAllowlist", ""));
```

Comment:
Should we have constants (e.g. probably in `ConfigKeys`) for these?

Reply:
I didn't want to pollute the main "ConfigKeys",
ideally AuthenticationService plugins are not hard coded into Venice, so I am not sure that it is good to have so many specific parameters in that class.

Maybe we can create a ConfigKeys class here in this package

Reply:
let's create constants